sktime/forecasting/model_selection/tests/test_split.py

```diff
@@ -711,3 +711,20 @@ def test_same_loc_splitter_hierarchical():
     for (t1, tt1), (t2, tt2) in zip(split_template_loc, split_templated_loc):
         assert np.all(t1 == t2)
         assert np.all(tt1 == tt2)
+
+
+@pytest.mark.parametrize("CV", [SlidingWindowSplitter, ExpandingWindowSplitter])
+@pytest.mark.parametrize("fh", [*TEST_FHS, *TEST_FHS_TIMEDELTA])
+@pytest.mark.parametrize("window_length", TEST_WINDOW_LENGTHS)
+@pytest.mark.parametrize("step_length", TEST_STEP_LENGTHS)
+def test_windowbase_splitter_get_n_split_hierarchical(
+    CV, fh, window_length, step_length
+):
+    """Test that WindowBaseSplitter.get_n_splits works for hierarchical data."""
```

Comment:
should we also look at unequal index sets to be sure?

Reply:
added here [d0dc47f](https://github.com/sktime/sktime/pull/5012/commits/d0dc47f329cce2b98f7c0aef62a60aeb1ce573f5). created another test to strike a balance between coverage and speed.